### ISSUE TITLE
CommandToolbarButtons no longer sync enabled state

### ISSUE THREAD

**Issue by timkpaine** [jupyterlab/jupyterlab#5740]
In latest JLab, looks like the `isEnabled` works correctly for the command palette but doesn't propagate this information to reenable buttons in the Notebook toolbar. A compact repro is available in this repo:

https://github.com/timkpaine/runall-extension

Here is the behavior:

![tmp](https://user-images.githubusercontent.com/3105306/49667388-e1e02880-fa28-11e8-84af-c51908ba4390.gif)

**Comment by jasongrout** [comment]
Did this used to work? This sort of issue precipitated our react overhaul in #5479 (see also #5475 and #5454)

**Comment by jasongrout** [comment]
This should be fixed by #5479. CC @saulshanabrook.

**Comment by timkpaine** [comment]
@jasongrout yep used to work fine. this code is still ugly from a user's perspective, icons seem to be sometimes working sometimes not (cc @vidartf who's kernel spy icons seem to be broken in the toolbar, along with several of my own extensions). If theres a "new" right way to do it im happy to adjust.

**Comment by vidartf** [comment]
@timkpaine The icons issue for the kernel-spy extensions recently got a fix here: https://github.com/vidartf/jupyterlab-kernelspy/pull/6 I still need to get around to testing that fully and doing a fresh release.

**Comment by timkpaine** [comment]
@vidartf yeah we hit that change as well, something to consider for @jasongrout @saulshanabrook @blink1073  @ellisonbg et al as this *feels* like something that will break again down the road

**Comment by saulshanabrook** [comment]
I think part of the issue is that we don't use any `CommandToolbarButton`s in core for the notebook, so we probably don't notice if they break. Lemme see if I can add a test for them.

**Comment by saulshanabrook** [comment]
I copied that extension into core to debug and I think I found the issue.

We currently ignore all command changes that don't have a type of `changed`, when deciding whether to re-render a command button: https://github.com/jupyterlab/jupyterlab/blob/9e980648319ad0a5252195e3639a69a294624785/packages/apputils/src/toolbar.tsx#L463-L465

However, it looks like we are also getting lot's of `many-changed` messages, which seem like they should also triggering re-rendering the button. When I re-render on this type, the button appears enabled. So the logic  for whether to re-render becomes:

```typescript
          (args.id == props.id && args.type == 'changed') ||
          args.type == 'many-changed'
```

Does this seem reasonable? When the type is `many-changed` I have observed the `id` is `undefined`. 

I am working on re-enabling the tests for this functionality on top of our recent React integration refactoring.
